File: backend/app/schemas/safety.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

SafetySourceType = Literal["journal_entry", "chat_message", "community_post", "community_comment", "ai_quiz_answer", "manual_check"]
SafetySeverity = Literal["low", "medium", "high", "crisis"]


class SafetyFlagResponse(BaseModel):
    id: int
    user_id: int | None
    source_type: str
    source_id: int | None
    severity: str
    status: str
    matched_signals: list[str]
    content_excerpt: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SafetyCheckRequest(BaseModel):
    text: str = Field(min_length=1, max_length=10000)
    source_type: SafetySourceType = "manual_check"


class SafetyCheckResponse(BaseModel):
    is_flagged: bool
    severity: SafetySeverity | None
    matched_signals: list[str]
    message: str | None


class CrisisResource(BaseModel):
    title: str
    description: str
    action_label: str
    action_value: str
    country: str = "US"

File: mobile/src/api/safety.ts
import { apiFetch } from './client';

export type CrisisResource = {
  title: string;
  description: string;
  action_label: string;
  action_value: string;
  country: string;
};

export type SafetyCheckResponse = {
  is_flagged: boolean;
  severity: 'low' | 'medium' | 'high' | 'crisis' | null;
  matched_signals: string[];
  message: string | null;
};

export function getCrisisResources() {
  return apiFetch<CrisisResource[]>('/safety/resources', { auth: false });
}

export function checkSafetyText(text: string) {
  return apiFetch<SafetyCheckResponse>('/safety/check', {
    method: 'POST',
    auth: true,
    body: JSON.stringify({ text, source_type: 'manual_check' }),
  });
}
